### PR TITLE
Avoid com.thoughtworks.xstream.annotations

### PR DESCRIPTION
**Pull request by jglick**
Compatibility with https://github.com/jenkinsci/jenkins/pull/4944.

### CODE REVIEW

**Comment by dkjellin** [comment]
Thank you for your PR, we will review this shortly 
sorry for the delay.

**Comment by jglick** [comment]
Also see the similar https://github.com/jenkinsci/atlassian-jira-software-cloud-plugin/pull/40, if you maintain that plugin as well.

File: src/main/java/com/atlassian/bitbucket/jenkins/internal/config/BitbucketPluginConfiguration.java [hunk 1]
```diff
@@ -40,6 +40,7 @@
     private List<BitbucketServerConfiguration> serverList = new ArrayList<>();
 
     public BitbucketPluginConfiguration() {
+        getConfigFile().getXStream().alias("atl-bbs-configuration", BitbucketPluginConfiguration.class);
```
Comment (dkjellin) [inline src/main/java/com/atlassian/bitbucket/jenkins/internal/config/BitbucketPluginConfiguration.java:43]: I think this is possibly in the wrong spot?
When I try an upgrade from current master to this one the builds won't run due to
> java.lang.ClassCastException: com.atlassian.bitbucket.jenkins.internal.config.BitbucketPluginConfiguration cannot be cast to com.atlassian.bitbucket.jenkins.internal.config.BitbucketServerConfiguration

**Review by dkjellin — CHANGES_REQUESTED** [review]
Changes are needed such that upgrades work

**Comment by dkjellin** [comment]
> Also see the similar [jenkinsci/atlassian-jira-software-cloud-plugin#40](https://github.com/jenkinsci/atlassian-jira-software-cloud-plugin/pull/40), if you maintain that plugin as well.

I have forwarded that to the team looking after that plugin. Thank you for raising it!

File: src/main/java/com/atlassian/bitbucket/jenkins/internal/config/BitbucketPluginConfiguration.java (hunk 1 above)
Comment (jglick) [reply src/main/java/com/atlassian/bitbucket/jenkins/internal/config/BitbucketPluginConfiguration.java:43]: Oops, fixing…

**Review by dkjellin — APPROVED** [review]

**Review by mhenschke-atl — APPROVED** [review]

**Pull request merged**